clients/ui/frontend/src/app/AppRoutes.tsx

```diff
@@ -0,0 +1,63 @@
+import * as React from 'react';
+import { Route, Routes } from 'react-router-dom';
+import { Dashboard } from '@app/Dashboard/Dashboard';
+import { Support } from '@app/Support/Support';
+import { NotFound } from '@app/NotFound/NotFound';
+
+export const isNavDataGroup = (navItem: NavDataItem): navItem is NavDataGroup => 'children' in navItem;
+
+type NavDataCommon = {
+  label: string;
+};
+
+export type NavDataHref = NavDataCommon & {
+  path: string;
+};
+
+export type NavDataGroup = NavDataCommon & {
+  children: NavDataHref[];
+};
+
+type NavDataItem = NavDataHref | NavDataGroup;
+
+export const useAdminSettings = (): NavDataItem[] => {
+  // get auth access for example set admin as true
+  const isAdmin = true //this should be a call to getting auth / role access
```

Comment:
You can also test that changing this to `false` will remove the fake Settings nav items

Reply:
mmm not quite what I expected, this is great to hide the settings for the user, but they can still navigate to those routes adding the path to the browser. 
A great example would be havving `Setting 1` as a new mock page called "Admin", and inside <Routes> either not mounting that route with conditional rendering or creating a private route.